addons/sale_crm/models/crm_lead.py

```diff
@@ -101,3 +101,10 @@ def _merge_get_fields_specific(self):
         # add all the orders from all lead to merge
         fields_info['order_ids'] = lambda fname, leads: [(4, order.id) for order in leads.order_ids]
         return fields_info
+
+    def _update_revenues_from_so(self):
+        for opportunity in self:
+            min_order_untaxed_total = opportunity.order_ids._filter_sales_order("amount_untaxed")
```

Comment:
It's a bit strange to use a read_group since we already call the method on the order_ids recordset.

So I'd say either we use a read_group on all opportunities in self, which would be nice if doable, or we use `_filter_sales_order` to just filter the orders with the domain presently inside of that method, and return the min value of the argument field among the filtered orders. (no read_group then)

Reply:
Indeed, we have SO browsing opportunity browsing SO calling a method on SO ... Would also move the code here as pointed out by @Odoonan